accepted/2025/dotnet-package-install.md

```diff
@@ -0,0 +1,175 @@
+It can also be used to specify a package source key that is found in a config file.
+* `--output-directory <path>`: Sets the directory where packages will be downloaded.
+The default is the global packages folder.
+* `--exclude-version`: Omits version number from the package folder name.
+* `--prerelase`: Allows pre-release packages to be considered when resolving versions.
+By default only stable versions are selected.
+* `--framework <tfm>`: Specifies the target framework.
+* `--download-only`: Downloads the `.nupkg` file to the output directory without extracting it.
+No dependencies are downloaded either.
+* `--no-http-cache`: Prevents the usage of http cached packages.
+* `--direct-download`: download directly without populating any caches with metadata.
+* `--configfile <path>`: Used to specify a NuGet.Config file to use instead of the default configuration hierarchy.
+* `--dependency-version <policy>`: Controls how versions of dependencies are selected when a version range is specified.
+Values: Lowest(default), HighestPatch, HighestMinor, and Highest.
+* `--no-audit`: opts out of vulnerability auditing. Be default it is on.
+* `--allow-insecure-connections`: allows downloading from http sources(insecure).
+
+### Technical explanation
+
+The `NuGet.CommandLine.XPlat` project currently exposes `dotnet package search` and `dotnet package update` in `Program.cs`.
+The upcoming `install` sub‑command fits naturally alongside these operations.
+
+### Algorithm: `dotnet package install <packageId> [options]`
+
+1. **Parse inputs**
+
+   * Extract `<packageId>` (required).
+   * Read all supported options:
+
+     * `--version <semver>`
+     * `--source <package source>`
+     * `--output-directory <path>`
+     * `--exclude-version`
+     * `--prerelease`
+     * `--framework <tfm>`
+     * `--download-only`
+     * `--no-http-cache`
+     * `--direct-download`
+     * `--configfile <path>`
+     * `--dependency-version <policy>`
+     * `--no-audit`
+     * `--allow-insecure-connections`
+
+1. **Load configuration**
+
+   * If `--configfile` is provided, load settings from that file.
+   * Otherwise, use the default `NuGet.Config` hierarchy.
+
+1. **Resolve package sources**
+
+   * If `--source` is specified, resolve those sources (it could be a url or a source key defined in a config file).
+   * Else, use all enabled sources from settings.
+
+1. **Http source usage check**
+
+    * If any of the sources are HTTP but do not have`allowInsecureConnections` set to true, log an error and stop the command execution.
+
+1. **Determine output directory**
+
+   * If `--output-directory` is specified, use it.
+   * Otherwise, default to the global packages folder.
+
+1. **Configure folder layout**
+
+   * If `--exclude-version` is set, package folders are named `<id>` only.
+   * Otherwise, use `<id>.<version>` layout.
+
+1. **Handle target framework**
+
+   * If `--framework` is provided, constrain resolution to that TFM.
```

Comment:
When a package has different dependencies for different TFMs, does specifying `--framework` mean only the dependencies for that specific TFM are downloaded?